src/WidgetBase.ts

```diff
@@ -549,12 +551,18 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 	 * @param dNode the dnode to process
 	 * @returns a VNode, string or null
 	 */
-	private dNodeToVNode(dNode: DNode): VNode | string | null {
+	private dNodeToVNode(dNode: DNode | DNode[]): (VNode | string | null)[] | VNode | string | null {
 
 		if (typeof dNode === 'string' || dNode === null) {
 			return dNode;
 		}
 
+		if (Array.isArray(dNode)) {
+			return dNode.map((node) => {
+				return this.dNodeToVNode(node) as VNode | string | null;
```

Comment:
Will always return a `VNode | string | null`, but the way the interface is need to cast - should probably add the correct overloads to make this play nicely.